components/proxy/src/main/java/com/hotels/styx/admin/handlers/ProxyStatusHandler.java

```diff
@@ -0,0 +1,52 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.admin.handlers;
+
+import com.hotels.styx.api.HttpRequest;
+import com.hotels.styx.api.HttpResponse;
+import com.hotels.styx.api.http.handlers.BaseHttpHandler;
+import com.hotels.styx.events.Event;
+import com.hotels.styx.events.EventNexus;
+import org.slf4j.Logger;
+
+import static com.hotels.styx.api.HttpResponse.Builder.response;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Handler that returns whether the proxy server has started.
+ */
+public class ProxyStatusHandler extends BaseHttpHandler {
+    private static final Logger LOGGER = getLogger(ProxyStatusHandler.class);
+
+    private volatile boolean proxyServerStarted;
+
+    public ProxyStatusHandler(EventNexus eventNexus) {
+        eventNexus.events("server.started.proxy")
+                .map(Event::value)
+                .cast(Boolean.class)
+                .subscribe(started -> {
+                    proxyServerStarted = started;
+                    LOGGER.info("proxyServerStarted={}", started);
+                });
```

Comment:
ProxyStatusHandler only needs to see the the proxy server state, which from `ProxyStatusHandler` point of view is just a read-only value.

Using an event bus for this use case is a bad practice due to:

1. It is a complicated way to communicate `proxyServerStarted` value to the handler.
2. The state management for "server.started.proxy" property is distributed to all listeners, instead of centralised for a single provider. This results in duplicated code: Every listener needs to implement the same 11 or 12 lines of code.
3. Opens up a possibility for data races. Consider what happens if server published the event "server.started.proxy" *before* some of the listeners was ready. That listener would miss the notification.

For these reasons, I would propose an opposing approach. Introduce a centralised configuration state variable for "server.started.proxy". Manage its state from the proxy server. Make it globally readable (read-only) via environment.

To clarify. I agree there definitely is a place for event based notifications within styx, but this endpoint is not one of them.

Reply:
I agree that feeding the endpoint directly with the events doesn't seem correct here.

Reply:
@mikkokar I understand your last two points in your list of three, but regarding your first, it is not clear - should we delay this feature until we have the "centralised configuration state variable[s]"? Otherwise it would require either: (1) using the existing `EventBus` which would seem to have the same issues, (2) implementing through some ad-hoc method which might be messy.

Reply:
Going forward (not necessary for this change) the `StyxConfigStore` will need to provide notification services along the lines of:

```
   styxConfg.watch("server.started.proxy")
       .subscribe(value -> { ... });
```